### DISCUSSION TITLE
Optional[FrameType] has no attribute "f_code"

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [python/typing#1217] [Q&A] [answered]
I want to identify the current executing function during runtime using the Python inspect() function (https://docs.python.org/3/library/inspect.html). The code  and mypy error are:
```
from types import FrameType
fn_name: str = inspect.currentframe().f_code.co_name
```
`error: Item "None" of "Optional[FrameType]" has no attribute "f_code"
`
Neither mypy reveal_type() and reveal_locals() functions show anything. What is the problem and solution? Thanks

**Comment by Gobot1234** [comment, marked as answer]
inspect.currentframe() can depending on the python implementation return None as stated in [the documentation](https://docs.python.org/3/library/inspect.html#inspect.currentframe):
> **CPython implementation detail:** This function relies on Python stack frame support in the interpreter, which isn’t guaranteed to exist in all implementations of Python. If running in an implementation without Python stack frame support this function returns `None`.

mypy is just letting you know that's the case. To handle this you can just check if the return of currentframe() is None
```py
from types import FrameType
frame = inspect.currentframe()
if frame is None:
    raise RuntimeError("unsupported python implementation")  # or give a default name
fn_name: str = frame.f_code.co_name
```

**Reply by Akuli** [reply]
I like to use `assert` for this:

```python3
frame = inspect.currentframe()
assert frame is not None
fn_name = frame.f_code.co_name
```
